src/pczt.rs

```diff
@@ -0,0 +1,406 @@
+}
+
+/// Information about an Orchars spend within a transaction.
+#[derive(Debug, Getters)]
+#[getset(get = "pub")]
+pub struct Spend {
+    /// The nullifier of the note being spent.
+    pub(crate) nullifier: Nullifier,
+
+    /// The randomized verification key for the note being spent.
+    pub(crate) rk: redpallas::VerificationKey<SpendAuth>,
+
+    /// The spend authorization signature.
+    ///
+    /// This is set by the Signer.
+    pub(crate) spend_auth_sig: Option<redpallas::Signature<SpendAuth>>,
+
+    /// The address that received the note being spent.
+    ///
+    /// - This is set by the Constructor (or Updater?).
+    /// - This is required by the Prover.
+    pub(crate) recipient: Option<Address>,
+
+    /// The value of the input being spent.
+    ///
+    /// - This is required by the Prover.
+    /// - This may be used by Signers to verify that the value matches `cv`, and to
+    ///   confirm the values and change involved in the transaction.
+    ///
+    /// This exposes the input value to all participants. For Signers who don't need this
+    /// information, or after signatures have been applied, this can be redacted.
+    pub(crate) value: Option<NoteValue>,
+
+    /// The rho value for the note being spent.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover.
+    ///
+    /// TODO: This could be merged with `rseed` into a tuple. `recipient` and `value` are
+    /// separate because they might need to be independently redacted. (For which role?)
+    pub(crate) rho: Option<Rho>,
+
+    /// The seed randomness for the note being spent.
+    ///
+    /// - This is set by the Constructor.
+    /// - This is required by the Prover.
+    pub(crate) rseed: Option<RandomSeed>,
+
+    /// The full viewing key that received the note being spent.
+    ///
+    /// - This is set by the Updater.
+    /// - This is required by the Prover.
+    pub(crate) fvk: Option<FullViewingKey>,
+
+    /// A witness from the note to the bundle's anchor.
+    ///
+    /// - This is set by the Updater.
+    /// - This is required by the Prover.
+    pub(crate) witness: Option<MerklePath>,
+
+    /// The spend authorization randomizer.
+    ///
+    /// - This is chosen by the Constructor.
+    /// - This is required by the Signer for creating `spend_auth_sig`, and may be used to
+    ///   validate `rk`.
+    /// - After`zkproof` / `spend_auth_sig` has been set, this can be redacted.
+    pub(crate) alpha: Option<pallas::Scalar>,
+
+    /// The ZIP 32 derivation path at which the spending key can be found for the note
+    /// being spent.
+    pub(crate) zip32_derivation: Option<Zip32Derivation>,
+
+    /// The spending key for this spent note, if it is a dummy note.
+    ///
+    /// - This is chosen by the Constructor.
+    /// - This is required by the IO Finalizer, and is cleared by it once used.
+    /// - Signers MUST reject PCZTs that contain `dummy_sk` values.
```

Comment:
- I both named this `dummy_sk` and added a "MUST reject" here because I do not want PCZTs to get serialized with non-dummy spending keys.
- I stored the dummy note's `sk` instead of the smaller-scoped `ask`, because in Orchard we impose validity requirements at parse time of `sk` (specifically to ensure that a valid `ivk` is produced), and we already had APIs for handling that parsing here. If desired this can be scoped down to `dummy_ask`, but given that (again) this field is only for dummy notes, I think `sk` should be fine here.